File: src/main/java/com/felipe/booking/app/dto/RoomAvailabilityDTO.java
package com.felipe.booking.app.dto;

import com.felipe.booking.domain.model.RoomAvailability;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;

@Data
@AllArgsConstructor
public class RoomAvailabilityDTO {
    private List<LocalDate> daysAvailable;
    private List<LocalDate> daysBooked;

    public static RoomAvailabilityDTO of(RoomAvailability roomAvailability) {
        return new RoomAvailabilityDTO(roomAvailability.getDaysAvailable(), roomAvailability.getDaysBooked());
    }
}

File: src/main/kotlin/com/felipe/booking/alten/app/dto/RoomAvailabilityDTO.kt
package com.felipe.booking.alten.app.dto

import com.felipe.booking.alten.domain.model.RoomAvailability
import java.time.LocalDate

data class RoomAvailabilityDTO(
    val daysAvailable: List<LocalDate>,
    val daysBooked: List<LocalDate>
) {
    companion object {
        fun of(roomAvailability: RoomAvailability) =
            RoomAvailabilityDTO(
                daysAvailable = roomAvailability.daysAvailable,
                daysBooked = roomAvailability.daysBooked
            )
    }
}